### ISSUE TITLE
[a11y] Modal pulled from payments needs improved a11y

### ISSUE THREAD

**Issue by data-sync-user** [mozilla/fxa#5016]
The `DialogMessage` from `fxa-payments-server` will be pulled in or the functionality will referenced for FXA-1621. However, going through this with a screenreader, there are a few issues that would be a11y wins if implemented:

* The modal should close on escape key press
* On modal close, shift focus back to the element that opened the modal
* When the modal is open, content behind it should not be read. Likely, the modal will need to be pulled out where it currently lies in the DOM, and placed adjacent to main page content. When the modal is active, the div for the main page content would then receive an `aria-hidden="true"` along with any other adjacent divs. When the modal is not active, it should have `aria-hidden="true"` and the main page content should _not_ have this property.
* To prevent scrolling in the background, we could add a class to the `body` (or container component of modal/page content) when the modal is open to set `overflow:hidden`. A dark background with opacity set (up to design) could be set here.
* Needs an `aria-labelledby="Modal_Header_ID"` and `aria-describedby="Modal_Content_ID`.
* Should have `aria-modal="true"` (this is new, do we want to use this?) or `role="dialog"` present.
* May need a tab fence once pulled out of the DOM (an empty div with a `tabindex="0"`) to ensure items inside are tabbable. Check for this last - go through the page with a screenreader and trigger a modal... can you tab around as expected, or does tab focus seem to get lost somewhere? If it does, try adding a tab fence.

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXA-1691)
┆Issue Number: FXA-1691
